lib/block.rb

```diff
@@ -0,0 +1,24 @@
+require 'date'
+
+module Hotel
+  class Block < Reservation
+    attr_reader :start_date, :end_date, :date_range, :disc_rate, :available_rooms, :reserved_rooms
+
+    def initialize(start_date, end_date, collection_of_rooms, disc_rate)
+      super(start_date,end_date)
+      @date_range = get_date_range(start_date, end_date)
+      @collection_of_rooms = collection_of_rooms
+      @disc_rate = disc_rate
+      @available_rooms = collection_of_rooms
+      @reserved_rooms = []
+    end
+
+    def reserve_room(room)
+      raise RuntimeError.new("room is not available in hotel block") unless @available_rooms.include?(room)
+
+      @available_rooms.delete(room)
+      @reserved_rooms.push(room)
```

Comment:
This is pretty good, but did you consider using the rooms as keys and a bool as a value in a hash? That would reduce your data overhead a bit, and means that you don't run the risk of forgetting to update one of the two arrays! :)